src/hooks/useSavedFilters.tsx

```diff
@@ -26,7 +26,10 @@ const useSavedFiltersQuery = () => {
 	);
 
 	const { mutate: updateSavedFilters, isLoading: isUpdating } = useMutation(
-		(data: { filter: SavedFilterType; onSuccess?: () => void }) => putSavedFilters(data.filter.filter_id, data.filter),
+		(data: { filter: SavedFilterType; onSuccess?: () => void }) => {
+			data.filter.query = _.omit(data.filter.query, 'filter_builder');
```

Comment:
we should do this only when filter builder exists.
Also add a comment why this line is added

Reply:
Addressed this.